backend/tracim_backend/views/core_api/account_controller.py

```diff
@@ -52,300 +21,32 @@
-        )
-        uapi.set_password(
-            request.current_user,
-            hapic_data.body.loggedin_user_password,
-            hapic_data.body.new_password,
-            hapic_data.body.new_password2,
-            do_save=True,
-        )
-        return
-
-    @hapic.with_api_doc(tags=[SWAGGER_TAG__ACCOUNT_CONTENT_ENDPOINTS])
-    @check_right(is_user)
-    @hapic.input_query(UserWorkspaceFilterQuerySchema())
-    @hapic.output_body(WorkspaceDigestSchema(many=True))
-    def account_workspace(self, context, request: TracimRequest, hapic_data=None):
-        """
-        Get list of auth user workspaces
-        """
-        app_config = request.registry.settings["CFG"]  # type: CFG
-        wapi = WorkspaceApi(
-            current_user=request.current_user, session=request.dbsession, config=app_config  # User
-        )
-        workspaces = wapi.get_all_for_user(
-            request.current_user,
-            include_owned=hapic_data.query.show_owned_workspace,
-            include_with_role=hapic_data.query.show_workspace_with_role,
-        )
-        return [wapi.get_workspace_with_context(workspace) for workspace in workspaces]
-
-    @hapic.with_api_doc(tags=[SWAGGER_TAG__ACCOUNT_CONTENT_ENDPOINTS])
-    @check_right(is_user)
-    @hapic.input_path(WorkspaceIdPathSchema())
-    @hapic.input_query(ActiveContentFilterQuerySchema())
-    @hapic.output_body(ContentDigestSchema(many=True))
-    def account_last_active_content(self, context, request: TracimRequest, hapic_data=None):
+    @hapic.input_path(PathSuffixSchema())
+    def account_route_put(self, context, request: TracimRequest, hapic_data: HapicData):
         """
-        Get last_active_content for user
+        Convenient route allowing to use PUT /api/v2/users/{user_id}/* endpoint with authenticated user
+        without giving directly user id.
+        This route generate a HTTP 307 with the right url
         """
-        app_config = request.registry.settings["CFG"]  # type: CFG
-        content_filter = hapic_data.query
-        api = ContentApi(
-            current_user=request.current_user, session=request.dbsession, config=app_config  # User
-        )
-        wapi = WorkspaceApi(
-            current_user=request.current_user, session=request.dbsession, config=app_config  # User
-        )
-        workspace = None
-        if hapic_data.path.workspace_id:
-            workspace = wapi.get_one(hapic_data.path.workspace_id)
-        before_content = None
-        if content_filter.before_content_id:
-            before_content = api.get_one(
-                content_id=content_filter.before_content_id,
-                workspace=workspace,
-                content_type=content_type_list.Any_SLUG,
+        return HTTPTemporaryRedirect(
+            request.url.replace(
+                "/users/me", "/users/{user_id}".format(user_id=request.current_user.user_id), 1
             )
```

Comment:
Still not convinced.

What about something like (seems more robust):

```suggestion
        redirection = "/users/{user_id}".format(user_id=request.current_user.user_id)
        if request.url = "/users/me":
            newurl = redirection
        else
           newurl = re.sub(r"^/users/me/", redirection + "/", request.url)
        return HTTPTemporaryRedirect(newurl, 1)
```

(:warning this suggestion requires importing the module `re`)

Reply:
Don't sure to understand your proposal, but this cannot work.
request.url does contain the full url, not just the ending path.